en/includes/deploy/security/encrypt-passwords-with-cipher-tool.md

```diff
@@ -0,0 +1,224 @@
+# Encrypt Passwords with the Cipher Tool
+
+The instructions on this page explain how plain text passwords in configuration files can be encrypted using the secure vault implementation that is built into the WSO2 Identity Server.  
+
+!!! info "Before you begin"
+    If you are using Windows, you need to have [**Ant**](http://ant.apache.org/) installed in order to use the Cipher Tool.
+
+## Encrypt passwords
+
+{% if product_name == "WSO2 Identity Server" and is_version != "7.0.0" %}
+
+While you are able to encrypt passwords using symmetric or asymmetric encryption, it is recommended to use symmetric encryption due to its greater resilience towards emerging post-quantum threats. 
+Asymmetric encryption methods like RSA are not recommended due to their vulnerability to quantum computing capabilities.
+
+!!! important
+    [Configure a separate keystore]({{base_path}}/deploy/security/keystores/configure-keystores/#configure-a-separate-keystore-for-encrypting-data-in-internal-datastores) as the internal keystore to encrypt passwords. If the internal keystore is not specified, the primary keystore will be used instead.
+
+Follow the steps below to encrypt passwords:
+
+1. Add the following `[secrets]` configurations to the bottom of the `deployment.toml` file in the `<IS_HOME>/repository/conf/` directory. Provide an alias for the password type followed by the actual password enclosed within square brackets `[]` as shown below.
+
+    ```toml
+    [secrets]
+    admin_password = "[password_1]"
+    keystore_password = "[password_2]"
+    key_password = "[password_3]"
+    truststrore_password = "[password_4]"
+    log4j2_password = "[password_5]"
+    ```
+
+2.  Open a terminal, navigate to the `<IS_HOME>/bin/` directory, and execute the following command based on the encryption method and your operating system:
+
+    === "Symmetric encryption"
+
+        !!! note "Prerequisite"
+
+            To support symmetric encryption, you should have a symmetric secret in an internal keystore of type PKCS12. Follow the instructions [here]({{base_path}}/deploy/security/keystores/configure-keystores/#add-a-symmetric-secret-to-a-pkcs12-keystore) to add one.
+
+        - **For Linux**: `./ciphertool.sh -Dconfigure -Dsymmetric`
+
+        - **For Windows**: `ciphertool.bat -Dconfigure -Dsymmetric`
+
+    === "Asymmetric encryption"
+
+        !!! warning
+
+            Asymmetric encryption methods like RSA are not recommended due to their vulnerability to post-quantum threats.
+
+        - **For Linux**: `./ciphertool.sh -Dconfigure`
+
+        - **For Windows**: `ciphertool.bat -Dconfigure`
+
+3. Open the `deployment.toml` again and see that the passwords are encrypted.
+
+    ```toml
+    [secrets]
+    admin_password = "xxxxxxx"
+    keystore_password = "xxxxxxx"
+    key_password = "xxxxxxx"
+    truststrore_password = "xxxxxxx"
+    "log4j.appender.LOGEVENT.password" = "xxxxxxx"
```

Comment:
```suggestion
    "log4j2_password" = "xxxxxxx"
```

Reply:
fixed with https://github.com/hwupathum/docs-is/pull/5/commits/3df8725bb45a9cfc4c10b1ae1c1237de5715212f